libs/wingc/src/type_check.rs

```diff
@@ -3472,6 +3472,15 @@ impl<'a> TypeChecker<'a> {
 		access: &AccessModifier,
 		env: &mut SymbolEnv,
 	) {
+		// Structs can't be defined in preflight or inflight contexts, only at the top-level of a program
+		if let Some(_) = env.parent {
+			self.spanned_error(
+				name,
+				format!(
+					"Structs must be declared at the top-level of a program: Struct '{name}' is not defined at the top-level"
```

Comment:
Avoiding repetitive wording, also changed program to file

```suggestion
					"struct {name} must be declared at the top-level of a file"
```